### DISCUSSION TITLE
Unreal Engine 4 implementation

### DISCUSSION THREAD

**Discussion by MaKiPL** [MaKiPL/OpenVIII-monogame#185] [General]
I'm currently undergoing OpenVIII implementation using Unreal Engine 4- only tests so far for mesh and texture rendering

Why?
1. Extremely easy modding for end-users
2. It's game engine, no more framework therefore we have all the optimizations and everything is based on shaders instead of CPU processing (although MonoGame used shaders on worldmap for vertices manipulation)
3. A lot of ready methods for programming
4. Cross-platform, big community
5. Extremely easy debugging (no need to integrate ImGui/ debug on our own)
6. Possibility of modding with newest standard graphics

Currently planned platforms:
1. Windows x64
2. Linux x64 (native)
3. Xbox One + Xbox Series X [already signed as UWP developer for Xbox consoles]
4. Android [that one is extremely easy to deploy from UE4]
5. iOS [never did that, but recently got iPhone]

**Comment by hogsy** [comment]
Just to share my two-cents, personally I'm not a huge fan of this switch over to using Unreal Engine 4 that a few open-source projects are making, because Unreal Engine 4 isn't under a typical license and you're left at the mercy of Epic Games. I'm happy to hear arguments to the contrary, but that's just how I see it. 

An important note here is the following ticket that's currently open; https://github.com/MaKiPL/OpenVIII-monogame/issues/172
To my understanding, GPLv3 is not compatible with Unreal Engine 4 as it violates the EULA. Again, I'm not a legal expert but the section I'm looking at is under [here](https://www.unrealengine.com/en-US/faq?active=source).

And in addition to that, Unreal Engine 4 is sort of a goliath to what I'd consider an already achievable project using any other existing open-source engine that are a fraction of the size (or rolling your own).

**Comment by Sebanisu** [comment]
https://answers.unrealengine.com/questions/277405/general-public-license-gpl-or-free-software.html yep GPL is not allowed.
